src/xenia/kernel/xboxkrnl/xboxkrnl_memory.cc

```diff
@@ -425,8 +425,12 @@ DECLARE_XBOXKRNL_EXPORT2(MmQueryAddressProtect, kMemory, kImplemented,
 
 void MmSetAddressProtect_entry(lpvoid_t base_address, dword_t region_size,
                                dword_t protect_bits) {
-  if (!protect_bits) {
-    XELOGE("MmSetAddressProtect: Failed due to incorrect protect_bits");
+  constexpr uint32_t required_protect_bits =
+      X_PAGE_NOACCESS | X_PAGE_READONLY | X_PAGE_READWRITE |
+      X_PAGE_EXECUTE_READ | X_PAGE_EXECUTE_READWRITE;
+
+  if (xe::bit_count(protect_bits & required_protect_bits) != 1) {
+    assert_false(xe::bit_count(protect_bits & required_protect_bits) > 1);
```

Comment:
The bit count may also be 0 (smaller than 1) — is that a special case that is correct? I think `assert_always` (which also accepts a description string) should be fine since the condition is already checked outside.

Reply:
That was conscious decision, such case (0 bits) is quite common and I don't want to spam assertions in that particular case.

Reply:
A comment about this case being common would be good for avoiding the same question in the future.